packages/web3-shared/src/hooks/useERC20TokensDetailedFromTokenLists.ts

```diff
@@ -11,40 +11,42 @@ import type { ERC20TokenDetailed } from '../types'
 export function useERC20TokensDetailedFromTokenLists(
     lists?: string[],
     keyword: string = '',
+    additionalTokens: ERC20TokenDetailed[] = [],
 ): AsyncStateRetry<ERC20TokenDetailed[]> {
     //#region fetch token lists
     const chainId = useChainId()
     const { fetchERC20TokensFromTokenLists } = useWeb3Context()
-    const { value: allTokens = [], ...asyncResult } = useAsyncRetry(
+    const { value: tokensFromList = [], ...asyncResult } = useAsyncRetry(
         async () => (!lists || lists.length === 0 ? [] : fetchERC20TokensFromTokenLists(lists, chainId)),
         [chainId, lists?.sort().join()],
     )
     //#endregion
-
     //#region fuse
     const fuse = useMemo(
         () =>
-            new Fuse(allTokens, {
+            new Fuse([...additionalTokens, ...tokensFromList], {
```

Comment:
What will happen if it got the same token both from token lists and DB?

Reply:
It's OK.
```
 const renderTokens = uniqBy([...tokens, ...filteredTokens, ...(searchedToken ? [searchedToken] : [])], (x) =>
        x.address.toLowerCase(),
    )
```